File: Java/src/main/java/QuestionBank/Solution1801_1900/Solution1893.java
package QuestionBank.Solution1801_1900;

import java.util.HashSet;

/**
 * @author windows
 */
public class Solution1893 {
    public boolean isCovered(int[][] ranges, int left, int right) {
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < ranges.length; i++) {
            for (int j = ranges[i][0]; j <= ranges[i][1]; j++) {
                if (j >= left && j <= right) {
                    set.add(j);
                }
            }
        }
        return set.size() == right - left + 1;
    }
    
    
    public static void main(String[] args) {
        Solution1893 solution1893 = new Solution1893();
    }
}

File: C++/src/Solution1801_1900/Solution1893.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution1893.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <algorithm>

bool Solution1893::isCovered(vector<vector<int>> &ranges, int left, int right) {
    sort(ranges.begin(), ranges.end());
    for (auto range : ranges) {
        if (range[0] <= left && range[1] >= left){
            left = range[1] + 1;
        }
    }
    return left > right;
}
